feat: add `shield:model` command for make custom `UserModel` (#491)
codeigniter4/shield: docs/concepts.md

You can use your own models to handle user persistence. Shield calls this the "User Provider" class. A default model
is provided for you at `CodeIgniter\Shield\Models\UserModel`. You can change this in the `Config\Auth->userProvider` setting.

> The only requirement is that your new class MUST extend the provided `UserModel`. Shield has a CLI command to quickly create a custom `UserModel`, for example if you run the following command in terminal:

Reviewer: ```suggestion
The only requirement is that your new class MUST extend the provided `UserModel`. Shield has a CLI command to quickly create a custom `MyUserModel` by running the following command in terminal:
```

```console
php spark shield:model MyUserModel
```

> You should set the `Config\Auth->userProvider` as below:

Reviewer: ```suggestion
You should set `Config\Auth::$userProvider` as follows:
```

Author: I think this phrase is used in several other places in the documents, I have to check the documents to correct them.

```php

> public $userProvider = 'App\Models\MyUserModel';

Reviewer: Use `::class` for better IDE and SA support.
```suggestion
public $userProvider = \App\Models\MyUserModel::class;
```

Author: Therefore, it should be changed in the following cases.

https://github.com/codeigniter4/shield/blob/96f9987d602138c5437f5d9f3d94503ef190eec1/src/Config/Auth.php#L60-L61

https://github.com/codeigniter4/shield/blob/96f9987d602138c5437f5d9f3d94503ef190eec1/src/Config/Auth.php#L343

Reviewer: Yes, that would be better.
